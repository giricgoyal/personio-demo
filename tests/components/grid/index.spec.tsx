import { render } from '@testing-library/react'
import React from 'react'
import Grid from 'src/components/grid'

describe('components/grid', () => {
    let container, onSortChangeHandler, onFilterChangeHandler
    const columnDef = [
        {
            title: 'Name',
            propBinding: 'name',
            filter: true,
            sort: true,
        },
    ]

    describe('When loading', () => {
        beforeEach(() => {
            onSortChangeHandler = jest.fn()
            onFilterChangeHandler = jest.fn()
            container = render(
                <Grid
                    data={[]}
                    isLoading
                    columnDef={columnDef}
                    title="sample grid"
                    sortBy="name"
                    filterBy={''}
                    onSortChange={onSortChangeHandler}
                    onFilterChange={onFilterChangeHandler}
                />,
            )
        })

        test('should render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When not loading', () => {
        const data = [
            {
                name: 'John',
                dateOfBirth: '2000-01-01',
            },
            {
                name: 'Jane',
                dateOfBirth: '1991-01-01',
            },
            {
                name: 'Ray',
                dateOfBirth: '1995-01-01',
            },
        ]

        beforeEach(() => {
            onSortChangeHandler = jest.fn()
            container = render(
                <Grid
                    data={data}
                    columnDef={columnDef}
                    title="sample grid"
                    sortBy="name"
                    filterBy={''}
                    onSortChange={onSortChangeHandler}
                    onFilterChange={onFilterChangeHandler}
                />,
            )
        })

        test('should render and match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })
})
