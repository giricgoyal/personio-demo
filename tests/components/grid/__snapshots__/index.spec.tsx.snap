// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/grid When loading should render and match snapshot 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="grid__header"
    >
      <div
        class="grid__header__left"
      >
        <div
          class="grid__header__left__title"
        >
          sample grid
        </div>
        <div
          class="grid__header__left__data-count"
        >
          (
          0
          /
          0
          )
        </div>
      </div>
      <div
        class="grid__header__right"
      >
        <div
          class="grid__header__right__filter"
        >
          <i
            class="fas fa-filter "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="filterby-select"
            >
              <option
                value="name"
              >
                Name
              </option>
            </select>
            <i
              class="fas fa-caret-down "
            />
          </label>
          <label
            class="combo-box"
          >
            <input
              data-testid="filter-value-input"
              list="filter-value-input"
              value=""
            />
          </label>
        </div>
        <div
          class="grid__header__right__sort"
        >
          <i
            class="fas fa-sort "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="select"
            >
              <option
                selected=""
                value="name"
              >
                Name
              </option>
            </select>
            <i
              class="fas fa-caret-down "
            />
          </label>
          <button
            class="button button--tertiary"
            data-testid="sort-order-button"
          >
            <i
              class="fas fa-arrow-up "
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="grid__loading"
    >
      <i
        class="fas fa-spinner fa-2x fa-spin"
      />
    </div>
  </div>
</div>
`;

exports[`components/grid When not loading When data is present should render and match snapshot 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="grid__header"
    >
      <div
        class="grid__header__left"
      >
        <div
          class="grid__header__left__title"
        >
          sample grid
        </div>
        <div
          class="grid__header__left__data-count"
        >
          (
          3
          /
          3
          )
        </div>
      </div>
      <div
        class="grid__header__right"
      >
        <div
          class="grid__header__right__filter"
        >
          <i
            class="fas fa-filter "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="filterby-select"
            >
              <option
                value="name"
              >
                Name
              </option>
            </select>
            <i
              class="fas fa-caret-down "
            />
          </label>
          <label
            class="combo-box"
          >
            <input
              data-testid="filter-value-input"
              list="filter-value-input"
              value=""
            />
          </label>
        </div>
        <div
          class="grid__header__right__sort"
        >
          <i
            class="fas fa-sort "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="select"
            >
              <option
                selected=""
                value="name"
              >
                Name
              </option>
            </select>
            <i
              class="fas fa-caret-down "
            />
          </label>
          <button
            class="button button--tertiary"
            data-testid="sort-order-button"
          >
            <i
              class="fas fa-arrow-up "
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="grid__table grid__table--grid-1"
    >
      <div
        class="grid__table__cell grid__table__cell--header"
      >
        <div
          class="grid__table__cell--header__label"
        >
          Name
        </div>
        <i
          class="fas fa-arrow-up "
        />
      </div>
      <div
        class="grid__table__cell"
      >
        Jane
      </div>
      <div
        class="grid__table__cell"
      >
        John
      </div>
      <div
        class="grid__table__cell"
      >
        Ray
      </div>
    </div>
  </div>
</div>
`;

exports[`components/grid When not loading When error occured should render and match snapshot 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="grid__header"
    >
      <div
        class="grid__header__left"
      >
        <div
          class="grid__header__left__title"
        >
          sample grid
        </div>
        <div
          class="grid__header__left__data-count"
        >
          (
          0
          /
          0
          )
        </div>
      </div>
      <div
        class="grid__header__right"
      >
        <div
          class="grid__header__right__filter"
        >
          <i
            class="fas fa-filter "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="filterby-select"
            >
              <option
                value="name"
              >
                Name
              </option>
            </select>
            <i
              class="fas fa-caret-down "
            />
          </label>
          <label
            class="combo-box"
          >
            <input
              data-testid="filter-value-input"
              list="filter-value-input"
              value=""
            />
          </label>
        </div>
        <div
          class="grid__header__right__sort"
        >
          <i
            class="fas fa-sort "
          />
          <label
            class="select"
          >
            <span
              class="select__label"
            />
            <select
              class="select__element"
              data-testid="select"
            >
              <option
                selected=""
                value="name"
              >
                Name
              </option>
            </select>
            <i
              class="fas fa-caret-down "
            />
          </label>
          <button
            class="button button--tertiary"
            data-testid="sort-order-button"
          >
            <i
              class="fas fa-arrow-up "
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="grid__error"
    >
      an error occured
      <div>
        <a
          data-testid="refresh-grid-button"
        >
          <i
            class="fas fa-sync fa-2x"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
